include/experimental/__p0009_bits/extents.hpp

```diff
@@ -251,12 +251,17 @@ struct maybe_static_array {
 
 #ifdef __cpp_lib_span
   MDSPAN_TEMPLATE_REQUIRES(class T, size_t N,
-                           /* requires */ (N == m_size_dynamic))
+                           /* requires */ (N == m_size_dynamic && N > 0))
```

Comment:
needed a separate constructor to make this work warning free, and make it work in constant expressions. Same strategy as with the from std::array ctor.